### DISCUSSION TITLE
Cannot read properties of undefined (reading 'when') on first run - Solid-Start in SSR

### DISCUSSION THREAD

**Discussion by wedwardbeck** [TanStack/query#6623] [General]
I am using solid-query in a Solid-Start SSR project and each time the query is first run, the page gets the error:
Cannot read properties of undefined (reading 'when')
If I refresh, it renders as expected.  I've searched Discord and here for any clues, but am finding nothing fits.  
Sample of repo with reproduced errors [here](https://codesandbox.io/p/github/wedwardbeck/sq-when/main?workspaceId=4f7a079f-40eb-4741-8ece-8dac4ed6e34e).  
Click link "solid-query issue" after home loads.  Get error.  Reload.  Data renders.  Same result on the table page .  
I was able to reproduce this on a CSR app as well, so I do not believe it is the SSR causing this.   
Can anyone provide some advice to correct this?

**Comment by aguilera51284** [comment]
i have the same problem :/
